internal/wasmldr/src/loader/configured/mod.rs

```diff
@@ -53,21 +52,39 @@ impl Loader<Configured> {
     }
 
     pub fn next(self) -> Result<Loader<Requested>> {
-        // TODO:
-        //   1. get attestation
-        //   2. get intermediate cert
-        //   3. choose key type based on technology
-        //   4. generate tech-specific attestation extension request
+        // TODO: get intermediate cert
+
+        let platform = Platform::get()?;
+        let cert_algo = match platform.technology() {
+            Technology::Snp => SECP_384_R_1,
+            _ => SECP_256_R_1,
+        };
 
         // Generate a keypair.
-        let raw = PrivateKeyInfo::generate(SECP_256_R_1)?;
+        let raw = PrivateKeyInfo::generate(cert_algo)?;
         let pki = PrivateKeyInfo::from_der(raw.as_ref())?;
 
+        let mut key_hash = vec![];
+
+         match platform.technology() {
+            Technology::Snp => {
+                let hash = Sha384::digest(raw.as_slice());
+                io::copy(&mut hash.as_slice(), &mut key_hash)?;
```

Comment:
Use [this](https://doc.rust-lang.org/std/primitive.slice.html#method.copy_from_slice) instead.